src/ducks/Studio/timeseries/hooks.js

```diff
@@ -56,6 +57,30 @@ function abortAllMetrics (abortables) {
   return [...abortables.values()].forEach(cancelQuery)
 }
 
+export const getPreparedMetricSettings = (metrics, settings) => {
+  const hasDaaMetric = metrics.includes(Metric.daily_active_addresses)
+
+  if (hasDaaMetric) {
+    metrics.forEach(metric => {
+      settings.set(metric, {
+        interval: '1d'
+      })
+    })
+  } else {
+    settings.forEach((value, key) => {
+      const newValue = {
+        ...value
+      }
+
+      delete newValue['delete']
+
+      settings.set(key, newValue)
+    })
+  }
+
+  return settings
+}
+
```

Comment:
1. This file is for hooks or functions that are directly related to the hook (e.g. `abortRemovedMetrics`);
2. Since `settings` is a map, variable name should be more descriptive, e.g `settingsMap`;
3. Better to return a new map, not to mutate the old one.